### PR TITLE
refactor(tokenselector): 3 - streamline token search results

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

Refactors `TokenSearchContent` to improve maintainability and performance as part of the token-selector component split work. The component is decomposed into focused modules, integrated with `VirtualList` for efficient rendering of large token lists, and includes comprehensive test coverage.

**Key Changes:**

- **Component Decomposition**: Split `TokenSearchContent` into smaller, single-responsibility modules:
  - `types.ts` - Centralized type definitions
  - `helpers.ts` - `appendImportSection` utility function
  - `useSearchRows.ts` - Hook for generating virtualized row data
  - `TokenSearchRowRenderer.tsx` - Row rendering component
  - `GuideBanner.tsx` - Extracted "Can't find your token?" banner component

- **VirtualList Integration**: Token search results now render through `VirtualList`, ensuring only visible items are hydrated and improving performance with large result sets. The "Can't find your token?" banner is now part of the virtualized list.

- **VirtualList Enhancements**: 
  - Updated `scrollResetKey` prop type from `string | number | boolean` to `number | string` for better type safety
  - Optimized rendering by replacing loop with `map` function
  - Fixed scroll reset bug by using `searchInput` directly instead of a hash, preventing collisions

- **Test Coverage**: Added comprehensive unit tests for `useSearchRows` hook (37 test cases) covering loading states, row generation, import sections, edge cases, and memoization behavior.

# To Test

1. General search flow
   - [ ] Open the token selector, type a search term with many matches, and scroll; the banner should stay pinned to the list and rendering should feel smooth.
   - [ ] Switch search terms rapidly—verify the list snaps back to the top and doesn't flash duplicates.

2. Import sections
   - [ ] Search for an address so blockchain/inactive/external sections appear; confirm section headers render once and each block has the expected top/bottom padding.
   - [ ] Use the "Import" buttons to ensure IDs/handlers still work.

3. Bridge context
   - [ ] With bridging enabled (pre-filtered tokens), confirm search results still render and virtualization doesn't break the filtered list.

# Background

Part of the token-selector component split work, focusing on improving code organization and performance before adding recents/favorites functionality.

## Summary by CodeRabbit

* **New Features**
  * Virtualized list rendering for faster, smoother token search results.

* **Style**
  * Removed the inline "Can't find your token" guide banner from the old results view.
  * Added section titles to better organize import token groups.
  * Adjusted spacing for import items so first/last entries render with correct padding.

* **Bug Fixes**
  * Token source tooltips now only appear when meaningful text is provided.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx [hunk 1]
```diff
@@ -58,59 +65,211 @@ export function TokenSearchContent({
...
+          shadowed={row.shadowed}
+          wrapperId={row.wrapperId}
+          isFirstInSection={row.isFirstInSection}
+          isLastInSection={row.isLastInSection}
+        />
+      )
+    default:
+      return null
+  }
+}
```
[the remark this answers is not in the document]
Comment (fairlighteth) [reply apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx]: @coderabbitai addressed in https://github.com/cowprotocol/cowswap/pull/6520/commits/03f600243b46e283a2772ea47c1068ff1cdfacf0

File: apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx [hunk 2]
```diff
@@ -58,59 +65,211 @@ export function TokenSearchContent({
...
+  const renderRow = useCallback(
+    (items: TokenSearchRow[], virtualItem: VirtualItem) => (
+      <TokenSearchRowRenderer
+        row={items[virtualItem.index]}
+        selectTokenContext={selectTokenContext}
+        importToken={importToken}
+      />
+    ),
+    [importToken, selectTokenContext],
+  )
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx]: Can it be a component instead of a callback that generates components?

Comment (fairlighteth) [reply apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx]: Please consider:
> VirtualList expects a (items, virtualItem) => ReactNode callback; TokenSearchRowRenderer is the component and renderRow is just the adapter. Changing it to a component would require changing VirtualList’s API, so I’d keep this as a callback unless we want to do that broader refactor.

**Review by limitofzero — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx [hunk 3]
```diff
@@ -58,59 +65,211 @@ export function TokenSearchContent({
...
+interface UseSearchRowsParams {
+  isLoading: boolean
+  matchedTokens: TokenWithLogo[]
+  activeList: TokenWithLogo[]
+  blockchainResult?: TokenWithLogo[]
+  inactiveListsResult?: TokenWithLogo[]
+  externalApiResult?: TokenWithLogo[]
+}
+
+function useSearchRows({
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx]: This pure component file contains too many mixed entities (other components, hooks, types), it must be decomposed

Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx]: This hook should be covered by unit-tests

Comment (fairlighteth) [reply apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx]: Addressed in https://github.com/cowprotocol/cowswap/pull/6520/commits/aa7fa8f9f901079ce02ecd835223ec538d003063

Comment (fairlighteth) [reply apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/index.tsx]: Added now in `apps/cowswap-frontend/src/modules/tokensList/pure/TokenSearchContent/useSearchRows.test.ts`

**Review by shoom3301 — APPROVED** [review]
